src/operations/create_collection.ts

```diff
@@ -96,31 +99,75 @@ export class CreateCollectionOperation extends CommandOperation<Collection> {
     session: ClientSession | undefined,
     callback: Callback<Collection>
   ): void {
-    const db = this.db;
-    const name = this.name;
-    const options = this.options;
+    (async () => {
+      const db = this.db;
+      const name = this.name;
+      const options = this.options;
 
-    const done: Callback = err => {
-      if (err) {
-        return callback(err);
+      const encryptedFields: Document | undefined =
+        options.encryptedFields ??
+        db.s.client.options.autoEncryption?.encryptedFieldsMap?.[`${db.databaseName}.${name}`];
+
+      if (encryptedFields) {
+        // Create auxilliary collections for FLE2 support.
+        const escCollection = encryptedFields.escCollection ?? `enxcol_.${name}.esc`;
+        const eccCollection = encryptedFields.eccCollection ?? `enxcol_.${name}.ecc`;
+        const ecocCollection = encryptedFields.ecocCollection ?? `enxcol_.${name}.ecoc`;
+
+        for (const collectionName of [escCollection, eccCollection, ecocCollection]) {
+          const createOp = new CreateCollectionOperation(db, collectionName);
+          await createOp.executeWithoutEncryptedFieldsCheck(server, session);
+        }
+      }
+
+      const coll = await this.executeWithoutEncryptedFieldsCheck(server, session);
+
+      if (encryptedFields) {
+        // Create the required index for FLE2 support.
+        const createIndexOp = new CreateIndexOperation(db, name, { __safeContent__: 1 }, {});
```

Comment:
Heads up: This may need to become a unique index to avoid easy-to-make user errors based on the slack conversation in https://mongodb.slack.com/archives/C0316KB5VM4/p1651503928461139?thread_ts=1651261523.846079&cid=C0316KB5VM4

Reply:
Should we do this now or wait until it is spec'ed. I think it's pretty much assured we need to set unique. Is there a ticket to block on?

Reply:
@nbbeeken @durran … and we decided not to do this since it would not work on sharded setups :slightly_smiling_face: So let’s stick to the spec here for now. We’ll probably have an ongoing conversation about this since I do think it’s a footgun, but this probably isn’t the solution.

Reply:
I thought so too! But yeah, makes sense that this doesn’t work :)